apps/api/src/app/sessions/sessions.service.ts

```diff
@@ -46,34 +90,29 @@ export class SessionsService {
     });
   }
 
-  async update(id: number, updateSessionInput: UpdateSessionInput): Promise<Session> {
-    try {
-      await this.sessionsRepository.update(id, {
-        name: updateSessionInput.name,
-        description: updateSessionInput.description,
-        status: updateSessionInput.status,
-        start: updateSessionInput.start,
-        end: updateSessionInput.end,
-        editable: updateSessionInput.editable,
-        enablePlayer: updateSessionInput.enablePlayer,
-        displaySampleSolution: updateSessionInput.displaySampleSolution,
-        enableLiveAnalysis: updateSessionInput.enableLiveAnalysis,
-        ...(updateSessionInput.mediaId && { media: { id: updateSessionInput.mediaId } }),
-      });
-
-      return this.findOne(id);
-    } catch (error) {
-      throw new BadRequestException(error.detail);
-    }
-  }
+  async update(id: number, updateSessionInput: UpdateSessionInput, owner: User): Promise<Session> {
+    const session = await this.findOneOfOwner(id, owner);
 
-  async remove(id: number) {
-    const deleteResult = await this.sessionsRepository.softDelete(id);
+    this.sessionsRepository.merge(session, {
+      name: updateSessionInput.name,
+      description: updateSessionInput.description,
+      status: updateSessionInput.status,
+      start: updateSessionInput.start,
+      end: updateSessionInput.end,
+      editable: updateSessionInput.editable,
+      enablePlayer: updateSessionInput.enablePlayer,
+      displaySampleSolution: updateSessionInput.displaySampleSolution,
+      enableLiveAnalysis: updateSessionInput.enableLiveAnalysis,
+      ...(updateSessionInput.mediaId && { media: { id: updateSessionInput.mediaId } }),
+    });
 
-    if (deleteResult.affected === 1) {
-      return this.findOne(id, true);
-    } else {
-      throw new NotFoundException();
-    }
+    return this.sessionsRepository.save(session);
+  }
+
+  async remove(id: number, owner: User) {
+    const session = await this.findOneOfOwner(id, owner);
+    await this.sessionsRepository.softRemove(session);
```

Comment:
Same here, that the session might not be found => not found exception?

Reply:
When it is null an NotFoundException is thrown by findONeOfOwner